src/construction.jl

```diff
@@ -29,15 +35,16 @@ function load_data(config::Dict, starttime::DateTime, endtime::DateTime)
     profile = filter(:id => in(ids), profile; view = true)
 
     # for forcing first get the right time range out
+    @assert issorted(forcing.time)
     startrow = searchsortedfirst(forcing.time, starttime)
     endrow = searchsortedlast(forcing.time, endtime)
     forcing = @view forcing[startrow:endrow, :]
     # then keep only IDs we use
     forcing = filter(:id => in(ids), forcing; view = true)
```

Comment:
Good catch that the assert must be moved up because of the searchsorted first.

In general in this function before I only validated the subset that we are using, and now we are validating the whole dataset. If it is cheap enough that is fine, but good to check the cost.

Reply:
The slowest is indeed validation the forcing on 3M rows, which takes 1.7s.

```julia
[ Info: Validating Legolas.SchemaVersion{Symbol("ribasim.node"), 1} for 29616 rows.
  0.019118 seconds (206.28 k allocations: 4.905 MiB)
┌ Warning: unsupported ARROW:extension:name type: "geoarrow.linestring", arrow type = Vector{Tuple{Float64, Float64}}
└ @ Arrow ~/.julia/packages/Arrow/QsQ3U/src/eltypes.jl:53
[ Info: Validating Legolas.SchemaVersion{Symbol("ribasim.edge"), 1} for 39043 rows.
  0.024101 seconds (426.03 k allocations: 10.671 MiB)
[ Info: Validating Legolas.SchemaVersion{Symbol("ribasim.state"), 1} for 8531 rows.
  0.028598 seconds (75.18 k allocations: 1.541 MiB, 76.58% gc time)
[ Info: Validating Legolas.SchemaVersion{Symbol("ribasim.static"), 1} for 5464 rows.
  0.010356 seconds (52.27 k allocations: 1.375 MiB)
[ Info: Validating Legolas.SchemaVersion{Symbol("ribasim.profile"), 1} for 46560 rows.
  0.029184 seconds (601.92 k allocations: 13.447 MiB)
[ Info: Validating Legolas.SchemaVersion{Symbol("ribasim.forcing"), 1} for 3088222 rows.
  1.741670 seconds (33.91 M allocations: 837.849 MiB, 4.59% gc time)
```

Reply:
Thanks for the numbers, I'd say 1.7s is no reason to worry for now, we don't have to optimize further. But perhaps later on if needed we will validate only what we use.